modules/wordpress-com-block-editor-iframe.php

```diff
@@ -0,0 +1,44 @@
+<?php
+/**
+ * Module Name: WordPress.com Block Editor Iframe
+ * Module Description: Allow new block editor posts to be composed on WordPress.com.
+ * Jumpstart Description: Allow new block editor posts to be composed on WordPress.com.
+ * Sort Order: 15
+ * First Introduced: 7.3
+ * Requires Connection: Yes
+ * Auto Activate: Yes
+ * Module Tags: Writing
+ * Feature: Writing
+ * Additional Search Queries: iframes, allow, compose, WordPress.com, block, editor, post
+ */
+
+function jetpack_disable_send_frame_options_header() {
+	if ( jetpack_framing_allowed() ) {
+		remove_action( 'admin_init', 'send_frame_options_header' );
+	}
+}
+add_action( 'admin_init', 'jetpack_disable_send_frame_options_header', 1 ); // High priority to get ahead of send_frame_options_header
+
+function jetpack_framing_allowed() {
+	if ( empty( $_GET['frame-nonce'] ) || false === strpos( $_GET['frame-nonce'], '.' ) ) {
+		return false;
+	}
+
+	list( $token, $signature ) = explode( '.', rawurldecode( $_GET['frame-nonce'] ) );
+
+	$verified = Jetpack::init()->verify_xml_rpc_signature( $token, $signature );
```

Comment:
Without the timestamp and nonce parameters, this will always return a `WP_Error`.

Since `WP_Error` objects are truthy, the following `if ( $verified ... )` conditional will always be true, and we'll always turn off the frame protection.

You can check this by requesting, for example `?frame-nonce={$token}.123` instead of `?frame-nonce={$token}.{$signature}`.

Reply:
Switching to another approach, see p9dueE-Gj-p2.